source/acquaman/IDEAS/IDEASXASScanActionController.cpp

```diff
@@ -232,7 +238,9 @@ AMAction3* IDEASXASScanActionController::createCleanupActions()
 {
 	AMListAction3 *cleanupActions = new AMListAction3(new AMListActionInfo3("IDEAS XAS Cleanup Actions", "IDEAS XAS Cleanup Actions"));
 
-	cleanupActions->addSubAction(IDEASBeamline::ideas()->scaler()->createDwellTimeAction3(0.1));
+	cleanupActions->addSubAction(new AMWaitAction(new AMWaitActionInfo(IDEASBeamline::ideas()->scaler()->dwellTime())));
```

Comment:
I thought @dretrex mentioned that this wasn't the only implementation that still needed strategic waits even after implementing the new continuous mode switching method?  I could be wrong.  But that would just support your suggestion that something needs to be tweaked in CLSSIS3820Scaler.